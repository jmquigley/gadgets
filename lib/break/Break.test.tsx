'use strict';

import {EnumValues as ev} from 'enum-values';
import {shallow} from 'enzyme';
import * as React from 'react';
import {Break, Sizing} from '../../dist/bundle';

for (const sizing of ev.getNames(Sizing)) {
	test('Test creation of a Break control of size ${sizing}', () => {
		const ctl = shallow(<Break sizing={Sizing[sizing]} />);

		expect(ctl).toBeDefined();
		expect(ctl).toMatchSnapshot();
	});
}
