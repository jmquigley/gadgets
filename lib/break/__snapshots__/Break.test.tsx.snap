// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a Break control of size \${sizing} 1`] = `
<styled.p
  className="ui-break"
  height="0.1875rem"
  sizing="xxsmall"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 2`] = `
<styled.p
  className="ui-break"
  height="0.25rem"
  sizing="xsmall"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 3`] = `
<styled.p
  className="ui-break"
  height="0.375rem"
  sizing="small"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 4`] = `
<styled.p
  className="ui-break"
  height="0.5rem"
  sizing="normal"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 5`] = `
<styled.p
  className="ui-break"
  height="0.5rem"
  sizing="medium"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 6`] = `
<styled.p
  className="ui-break"
  height="0.75rem"
  sizing="large"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 7`] = `
<styled.p
  className="ui-break"
  height="1rem"
  sizing="xlarge"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 8`] = `
<styled.p
  className="ui-break"
  height="1.5rem"
  sizing="xxlarge"
  style={Object {}}
/>
`;

exports[`Test creation of a Break control of size \${sizing} 9`] = `
<styled.p
  className="ui-break"
  height="0.5rem"
  sizing="inherit"
  style={Object {}}
/>
`;
